plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -73,7 +75,7 @@ type (
 )
 
 // NewMetricsReader returns a new MetricsReader.
-func NewMetricsReader(logger *zap.Logger, cfg config.Configuration) (*MetricsReader, error) {
+func NewMetricsReader(logger *zap.Logger, cfg config.Configuration, jt *jtracer.JTracer) (*MetricsReader, error) {
```

Comment:
By convention, observability components like logger and tracer are passed as last arguments. Since you're changing signature anyway, let's fix it.